### PR TITLE
Chore: rename contracts

### PR DESCRIPTION
**Pull request by haythem96**
# Task: rename contracts

## Description

Rename contracts:
- BullStrategy -> ZenBullStrategy
- AuctionBull -> ZenAuction
- FlashBull -> FlashZen
- LeverageBull -> LeverageZen
- EmergencyShutdown -> ZenEmergencyShutdown

## Type of change

- [ ] New feature
- [ ] Bug fix
- [ ] Testing code
- [X] Document update or config files

### CODE REVIEW

File: packages/bull-vault/src/ZenBullStrategy.sol [hunk 1]
```diff
@@ -89,8 +89,7 @@ contract BullStrategy is ERC20, LeverageBull {
         address _eulerMarketsModule
     )
         ERC20("Bull Vault", "BullVault")
```
Comment (aleone) [inline packages/bull-vault/src/ZenBullStrategy.sol:91]: This should be Zen Bull Vault and ZenBull imo.  I think this is the most important part to fix thb.

**Review by aleone — APPROVED** [review]
Missing change to ERC20 token name, imo thats most important part to do so should update it.

File: packages/bull-vault/script/Deploy.s.sol [hunk 2]
```diff
@@ -42,12 +42,12 @@ contract DeployScript is Script {
...
         // deploy contracts
-        bullStrategy =
-        new BullStrategy(crabAddress, powerTokenControllerAddress, eulerAddress, eulerMarketsModuleAddress);
-        emergencyShutdown = new EmergencyShutdown(address(bullStrategy), uniFactoryAddress);
-        auctionBull =
-        new AuctionBull(auctionManagerAddress, address(bullStrategy), uniFactoryAddress, crabAddress, eTokenAddress, dTokenAddress);
-        flashBull = new FlashBull(address(bullStrategy), uniFactoryAddress);
+        zenBullStrategy =
+        new ZenBullStrategy(crabAddress, powerTokenControllerAddress, eulerAddress, eulerMarketsModuleAddress);
+        emergencyShutdown = new ZenEmergencyShutdown(address(zenBullStrategy), uniFactoryAddress);
```
Comment (alpinechicken) [inline packages/bull-vault/script/Deploy.s.sol:47]: call it zenEmergencyShutdown here just for consistency?

**Review by alpinechicken — APPROVED** [review]
Lgtm

We still have a bunch of non-zen bull vars (not suggesting to change, just noting):

FlashWithdrawBullData 
bullToRedeem
bullStrategy
bullShare
data.bullToRedeem

but

**Comment by haythem96** [comment]
> Lgtm
> 
> We still have a bunch of non-zen bull vars (not suggesting to change, just noting):
> 
> FlashWithdrawBullData bullToRedeem bullStrategy bullShare data.bullToRedeem
> 
> but

Yeah, I think the vars name are not important, they are still relevant name wise.
Contracts name I guess are more important in case we will make a real bull strategy in the same monorepo

**Pull request merged**